storage/remote/otlptranslator/prometheus/normalize_name_test.go

```diff
@@ -88,33 +88,17 @@ func TestAllowUTF8(t *testing.T) {
 		require.Equal(t, "unsupported.metric.weird_+=.:,!* & #", normalizeName(createGauge("unsupported.metric.weird", "+=.:,!* & #"), "", true))
 		require.Equal(t, "unsupported.metric.redundant___test $_per_°C", normalizeName(createGauge("unsupported.metric.redundant", "__test $/°C"), "", true))
 		require.Equal(t, "metric_with_字符_foreign_characters_ど", normalizeName(createGauge("metric_with_字符_foreign_characters", "ど"), "", true))
+		require.Equal(t, "metric....split_=+by_//utf8characters", normalizeName(createGauge("metric....split_=+by_//utf8characters", ""), "", true))
 	})
 	t.Run("disallow UTF8", func(t *testing.T) {
 		require.Equal(t, "unsupported_metric_temperature_F", normalizeName(createGauge("unsupported.metric.temperature", "°F"), "", false))
 		require.Equal(t, "unsupported_metric_weird", normalizeName(createGauge("unsupported.metric.weird", "+=.:,!* & #"), "", false))
 		require.Equal(t, "unsupported_metric_redundant_test_per_C", normalizeName(createGauge("unsupported.metric.redundant", "__test $/°C"), "", false))
-		require.Equal(t, "metric_with_foreign_characters", normalizeName(createGauge("metric_with_字符_foreign_characters", "ど"), "", false))
+		require.Equal(t, "metric_with____foreign_characters", normalizeName(createGauge("metric_with_字符_foreign_characters", "ど"), "", false))
```

Comment:
The classical name translation mode, i.e. only allowing alphanumerics and colons, should _not_ change.

Reply:
Well, I understood that we would keep the number of characters at least and the previous behavior was a bug. The same way you did in your PR that allowed colons, you changed the test because it didn't reflect the behavior we wanted.

What I'm trying to avoid by at least keeping the same amount of characters is that we avoid collisions like this:

| Original metric       | Keeping same amount of characters (no name collision)| Keeping the current behavior (name collision)|
|-----------------------|-----------------------------------|------------------------------|
| http..requests..total | http__requests__total             | http_requests_total          |
| http.requests.total   | http_requests_total               | http_requests_total          |


I have no strong opinions though, I can see how this would be considered a breaking change, but we could also call this a bug fix 🤔

Reply:
> Well, I understood that we would keep the number of characters at least and the previous behavior was a bug.

When was that ever discussed? The bug that was [pointed out by me](https://github.com/prometheus/prometheus/pull/15258#discussion_r1832772051) is that the new UTF8 translation scheme translates some UTF8 characters to underscores, even though the scheme promises to not escape any UTF8 characers: "NoUTF8EscapingWithSuffixes will keep UTF-8 characters as they are". Please keep this PR on topic with that particular bug, and don't change the classical translation mode.

> The same way you did in your PR that allowed colons, you changed the test because it didn't reflect the behavior we wanted.

It's not the same because, A) it was on topic for the PR, B) not allowing colons in Prometheus metric names was a bug since the Prometheus metric name specification allows for colons.

Reply:
Ok no problems, I'll update the PR soon